src/main/java/ch/epfl/biop/operetta/companion/CompanionFileGenerator.java

```diff
@@ -0,0 +1,436 @@
+ *          this list of conditions and the following disclaimer
+ *          in the documentation and/or other materials provided with the distribution.
+ *      3. Neither the name of the copyright holder nor the names of its contributors
+ *          may be used to endorse or promote products
+ *          derived from this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDERS OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ *
+ * Ressources :
+ * - https://www.openmicroscopy.org/Schemas/Documentation/Generated/OME-2016-06/ome_xsd.html#BinaryFile
+ * - https://www.javatpoint.com/how-to-read-xml-file-in-java
+ * - https://stackoverflow.com/questions/11560173/how-can-i-read-xml-attributes-using-java
+ */
+
+
+import java.io.ByteArrayOutputStream;
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.OutputStream;
+import java.io.OutputStreamWriter;
+import java.io.UnsupportedEncodingException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import javax.xml.parsers.DocumentBuilder;
+import javax.xml.parsers.DocumentBuilderFactory;
+import javax.xml.transform.OutputKeys;
+import javax.xml.transform.Result;
+import javax.xml.transform.Source;
+import javax.xml.transform.Transformer;
+import javax.xml.transform.TransformerException;
+import javax.xml.transform.TransformerFactory;
+import javax.xml.transform.dom.DOMSource;
+import javax.xml.transform.stream.StreamResult;
+
+import ome.xml.model.Instrument;
+import ome.xml.model.MapAnnotation;
+import ome.xml.model.PlateAcquisition;
+import ome.xml.model.TagAnnotation;
+import org.w3c.dom.Document;
+import org.w3c.dom.Element;
+
+import ome.xml.model.Image;
+import ome.xml.model.OME;
+import ome.xml.model.Plate;
+import ome.xml.model.StructuredAnnotations;
+import ome.xml.model.Well;
+import ome.xml.model.WellSample;
+import ome.xml.model.primitives.NonNegativeInteger;
+
+/**
+ * The purpose of this class is to generate a .companion.ome file to create an OMERO plate object,
+ * with ONLY one image per well. This .companion.ome file can be directly uploaded on OMERO via
+ * OMERO.insight software, with screen import option.
+ *
+ */
+public class CompanionFileGenerator {
```

Comment:
Does this class already exist somewhere else ?

Reply:
No, made on purpose